lib/openstudio-standards/schedules/information.rb

```diff
@@ -40,14 +374,232 @@ def self.schedule_ruleset_get_min_max(schedule_ruleset)
+        year_end_date = OpenStudio::Date.new(OpenStudio::MonthOfYear.new('December'), 31, 2009)
+      end
+
+      # Get the ordered list of all the day schedules
+      day_schs = schedule_ruleset.getDaySchedules(year_start_date, year_end_date)
+
+      # Get the array of which schedule is used on each day of the year
+      day_schs_used_each_day = schedule_ruleset.getActiveRuleIndices(year_start_date, year_end_date)
+
+      # Create a map that shows how many days each schedule is used
+      day_sch_freq = day_schs_used_each_day.group_by { |n| n }
+
+      # Build a hash that maps schedule day index to schedule day
+      schedule_index_to_day = {}
+      day_schs.each_with_index do |day_sch, i|
+        schedule_index_to_day[day_schs_used_each_day[i]] = day_sch
+      end
+
+      # Loop through each of the schedules that is used, figure out the
+      # full load hours for that day, then multiply this by the number
+      # of days that day schedule applies and add this to the total.
+      annual_flh = 0.0
+      max_daily_flh = 0.0
+      default_day_sch = schedule_ruleset.defaultDaySchedule
+      day_sch_freq.each do |freq|
+        sch_index = freq[0]
+        number_of_days_sch_used = freq[1].size
+
+        # Get the day schedule at this index
+        day_sch = nil
+        day_sch = if sch_index == -1 # If index = -1, this day uses the default day schedule (not a rule)
+                    default_day_sch
+                  else
+                    schedule_index_to_day[sch_index]
+                  end
+        daily_flh = OpenstudioStandards::Schedules.schedule_day_get_equivalent_full_load_hours(day_sch)
+
+        # Multiply the daily EFLH by the number
+        # of days this schedule is used per year
+        # and add this to the overall total
+        annual_flh += daily_flh * number_of_days_sch_used
+      end
+
+      # Warn if the max daily EFLH is more than 24,
+      # which would indicate that this isn't a fractional schedule.
+      if max_daily_flh > 24
+        OpenStudio.logFree(OpenStudio::Warn, 'openstudio.standards.Schedules.Information', "#{schedule_ruleset.name} has more than 24 EFLH in one day schedule, indicating that it is not a fractional schedule.")
+      end
+
+      return annual_flh
+    end
+
+    # Returns an array of average hourly values from a ScheduleRuleset object
+    # Returns 8760 values, 8784 for leap years.
+    #
+    # @param schedule_ruleset [OpenStudio::Model::ScheduleRuleset] OpenStudio ScheduleRuleset object
+    # @return [Array<Double>] Array of hourly values for the year
+    def self.schedule_ruleset_get_hourly_values(schedule_ruleset)
+      # define the start and end date
+      year_start_date = nil
+      year_end_date = nil
+      if schedule_ruleset.model.yearDescription.is_initialized
+        year_description = schedule_ruleset.model.yearDescription.get
+        year = year_description.assumedYear
```

Comment:
We actually have an internal discussion on the assumedYear. Does this assumedYear guaranteed a non-leap year?
The reason is the PRM guideline requires PRM run in a non-leap year, so we may need some functions to make sure we are not running in leap year simulation.

Reply:
@weilixu

It does not guarantee a non-leap year. How about at the start of the PRM methods, adding a check for the year, and setting to a default non-leap year if it is a leap year?

Reply:
@weilixu looking more at the PRM code, there are a few places that use `schedule_get_hourly_values`. It seems those places specifically loop through weeks and hours to get exactly 8760 values, [get_fan_hours_per_week](https://github.com/NREL/openstudio-standards/blob/c1fbb2ff55b05b49599d892472e47c615c37e504/test/90_1_prm/prm_test_helper.rb#L2247) for example:
```
  def get_fan_hours_per_week(model, air_loop)
    fan_schedule = air_loop.availabilitySchedule
    fan_hours_8760 = OpenstudioStandards::Schedules.schedule_get_hourly_values(fan_schedule)
    fan_hours_52 = []

    hr_of_yr = -1
    (0..51).each do |iweek|
      week_sum = 0
      (0..167).each do |hr_of_wk|
        hr_of_yr += 1
        week_sum += fan_hours_8760[hr_of_yr]
      end
      fan_hours_52 << week_sum
    end
    max_fan_hours = fan_hours_52.max
    return max_fan_hours
  end

I'm wondering if we need to make any changes if the PRM methods that use the methods all force 8760 values anyways.
```

Reply:
to be safe, it seems the best way forward is to fix the year to a non-leap year if the AssumedYear is a leap year.

Reply:
@weilixu https://github.com/NREL/openstudio-standards/pull/1660